### PR TITLE
add StreamConformanceTest for HTTP CONNECT stream

### PR DESCRIPTION
**Pull request by geoffkizer**
@stephentoub @dotnet/ncl

### CODE REVIEW

File: src/libraries/System.Net.Http/tests/FunctionalTests/ResponseStreamConformanceTests.cs [hunk 1]
```diff
@@ -35,6 +35,26 @@ protected override async Task<StreamPair> CreateConnectedStreamsAsync()
...
+        }
+
+        protected override async Task<StreamPair> CreateConnectedStreamsAsync()
+        {
+            StreamPair pair = await base.CreateConnectedStreamsAsync();
+            Assert.True(pair.Stream2.CanWrite);
+            Assert.True(pair.Stream2.CanRead);
+            return pair;
+        }
+    }
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/tests/FunctionalTests/ResponseStreamConformanceTests.cs:56]: Is this testing a different stream than Http1UpgradeResponseStreamConformanceTests is testing?

Comment (geoffkizer) [reply src/libraries/System.Net.Http/tests/FunctionalTests/ResponseStreamConformanceTests.cs:56]: It's the same stream, but a different way of constructing it.

Comment (stephentoub) [reply src/libraries/System.Net.Http/tests/FunctionalTests/ResponseStreamConformanceTests.cs:56]: From a "we know the implementation" perspective, this doesn't seem like it's actually testing anything further: 
```C#
                else if (ReferenceEquals(normalizedMethod, HttpMethod.Connect) && response.StatusCode == HttpStatusCode.OK)
                {
                    responseStream = new RawConnectionStream(this);
                    _connectionClose = true; 
                }
                else if (response.StatusCode == HttpStatusCode.SwitchingProtocols)
                {
                    responseStream = new RawConnectionStream(this);
                }
```
but if you really think it's worth adding additional stream tests for, ok.

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Net.Http/tests/FunctionalTests/ResponseStreamConformanceTests.cs (hunk 1 above)
Comment (geoffkizer) [reply src/libraries/System.Net.Http/tests/FunctionalTests/ResponseStreamConformanceTests.cs:56]: Yeah, I think you're right here. I'll just close this.

**Pull request closed without merging**